internal/datastore/memdb/revisions.go

```diff
@@ -42,8 +41,8 @@ func (mdb *memdbDatastore) newRevisionID() revisions.TimestampRevision {
 func (mdb *memdbDatastore) HeadRevision(_ context.Context) (datastore.Revision, error) {
 	mdb.RLock()
 	defer mdb.RUnlock()
-	if mdb.db == nil {
-		return nil, fmt.Errorf("datastore has been closed")
+	if err := mdb.checkNotClosed(); err != nil {
```

Comment:
Add this check to SnapshotReader and ReadWriteTx as well

Reply:
`ReadWriteTx` already has it, and `SnapshotReader` doesn't access the `db` field.

Reply:
But trying to get a `SnapshotReader` on a closed memdb should also error

Reply:
Makes sense, added